### ISSUE TITLE
Inquiry Regarding Text Input Feature in SynTalker

### ISSUE THREAD

**Issue by Cal-DCosta** [RobinWitch/SynTalker#2]
Hi Robin,
I wanted to express my appreciation for your work on SynTalker;  it’s a app/project I’ve been searching for quite a while.
I have a question regarding the text input functionality mentioned in your paper. While I explored the Hugging Face demo, I was unable to locate an input box for text alongside the audio file input. Could you clarify how the text input is integrated into the application? Is it available in the demo, or is it implemented in a different part of the codebase?
Thank you for your time, and I look forward to your response.

**Comment by RobinWitch** [comment]
Thank you for paying attention to our work.
Now it only supports pure speech audio input. Enable text input simultaneously with speech is in schedule.
You can see our release plan. `Our syntalker can recieve both speech and text input simultaneously`. I will release it soon.
Because I am a bit busy lately and have some other things to do.
I think I will release sampling code in two weeks, online web demo and training code in three months.
Hope you can understand!
If it's urgent and you need to reproduce it yourself, feel free to contact with me.

**Comment by Cal-DCosta** [comment]
Thank you looking forward to the full demo, all the best!

**Comment by SreehariRajan** [comment]
Is the training script with text input available?

**Comment by RobinWitch** [comment]
Available.
Due to being occupied with other matters and the many dependencies involved in text inference, the progress of open source has been somewhat delayed. I plan to organize and release the inference code within a month, while the training code will be released within two months.

> Is the training script with text input available?

**Comment by SreehariRajan** [comment]
So currently in the repo only speech to motion training script is present ?

**Comment by RobinWitch** [comment]
Yes

**Comment by SreehariRajan** [comment]
Is the motion auto-encoder trained with BEATX and HumanML3D datasets for the given pre-trained weights ?

**Comment by RobinWitch** [comment]
> Is the motion auto-encoder trained with BEATX and HumanML3D datasets for the given pre-trained weights ?

Yes, we have provided it in this update. Please check [Enabling Text Control](https://github.com/RobinWitch/SynTalker/blob/1cd55d6be444bdc01485b93a2fcfb746dc145a19/README.md?plain=1#L110).

**Comment by SreehariRajan** [comment]
Is the code for visualizing with text prompt available ?

**Comment by RobinWitch** [comment]
> Is the code for visualizing with text prompt available ?

Yes! Now it's available, please checkout our [latest update](https://github.com/RobinWitch/SynTalker/blob/eb6e86c06410a01c3f7010eef2a9540e05c03c0d/README.md?plain=1#L129).

**Comment by SreehariRajan** [comment]
fatal: No url found for submodule path 'libs/human_body_prior/body_visualizer' in .gitmodules
Failed to recurse into submodule path 'libs/human_body_prior'

Can you add the URL ?

**Comment by RobinWitch** [comment]
> fatal: No url found for submodule path 'libs/human_body_prior/body_visualizer' in .gitmodules Failed to recurse into submodule path 'libs/human_body_prior'
> 
> Can you add the URL ?

Sorry, I forgot it, now I have added it !

**Comment by SreehariRajan** [comment]
Running the command `python preprocess/download_hf.py` downloads the dataset "beatx_1-30_amass_h3d." Does this include the complete AMASS dataset for training?

**Comment by RobinWitch** [comment]
> Running the command `python preprocess/download_hf.py` downloads the dataset "beatx_1-30_amass_h3d." Does this include the complete AMASS dataset for training?

No, it only includes preprocessed BEATX datasets (which also named BEAT2, include speaker 1-30), using AMASS data for training will additionally need another files, I will upload it to huggingface tomorrow. After downloading it, you need to set `use_amass` in `configs/diffusion_h3d.yaml` to `True` for using AMASS data in training.

I'm so busy recently. I think I will release preprocess raw data code in two weeks. Thank you for your attention and patience.

**Comment by SreehariRajan** [comment]
Ok, thank you,
The dataset you will upload in huggingface will be the dataset with corrected fps ?

**Comment by RobinWitch** [comment]
Yes, all will be right.

**Comment by SreehariRajan** [comment]
OK, thank you.
I hope you will also add the instructions for training in the README.

**Comment by SreehariRajan** [comment]
> > Running the command `python preprocess/download_hf.py` downloads the dataset "beatx_1-30_amass_h3d." Does this include the complete AMASS dataset for training?
> 
> No, it only includes preprocessed BEATX datasets (which also named BEAT2, include speaker 1-30), using AMASS data for training will additionally need another files, I will upload it to huggingface tomorrow. After downloading it, you need to set `use_amass` in `configs/diffusion_h3d.yaml` to `True` for using AMASS data in training.
> 
> I'm so busy recently. I think I will release preprocess raw data code in two weeks. Thank you for your attention and patience.

Is the dataset uploaded?

**Comment by RobinWitch** [comment]
> > > Running the command `python preprocess/download_hf.py` downloads the dataset "beatx_1-30_amass_h3d." Does this include the complete AMASS dataset for training?
> > 
> > 
> > No, it only includes preprocessed BEATX datasets (which also named BEAT2, include speaker 1-30), using AMASS data for training will additionally need another files, I will upload it to huggingface tomorrow. After downloading it, you need to set `use_amass` in `configs/diffusion_h3d.yaml` to `True` for using AMASS data in training.
> > I'm so busy recently. I think I will release preprocess raw data code in two weeks. Thank you for your attention and patience.
> 
> Is the dataset uploaded?

You need to wait for 1-3 days. My current network cannot directly connect to Huggingface, so I need to use a VPN. I have been experiencing frequent interruptions due to VPN network during uploads. I am currently trying to find a more stable VPN with better network conditions.

**Comment by SreehariRajan** [comment]
Ok, Thank you.
I hope you will also add the instructions for training in the README.

**Comment by RobinWitch** [comment]
> Ok, Thank you. I hope you will also add the instructions for training in the README.

I have uploaded it, please check.

**Comment by SreehariRajan** [comment]
Ok, thank you
